src/mongocxx/instance.cpp

```diff
@@ -96,7 +100,15 @@ class instance::impl {
         if (_user_logger) {
             libmongoc::log_set_handler(null_log_handler, nullptr);
         }
+
+        // Under ASAN, we don't want to clean up libmongoc, because it causes libraries to become
+        // unloaded, and then ASAN sees non-rooted allocations that it consideres leaks. These are
+        // also inscrutable, because the stack refers into an unloaded library, which ASAN can't
+        // report. Note that this only works if we have built mongoc so that it doesn't do its
+        // unfortunate automatic invocation of 'cleanup'.
+#if !__has_feature(address_sanitizer)
```

Comment:
I'm curious if there are any related C driver feature requests we could make, such that we would be able to get rid of this `#if` were they to be implemented.